sable_network/src/history/local_service.rs

```diff
@@ -103,20 +105,47 @@ impl<'a> LocalHistoryService<'a> {
         }
 
         if target_exists {
+            use crate::network::update;
+
             // "The order of returned messages within the batch is implementation-defined, but SHOULD be
             // ascending time order or some approximation thereof, regardless of the subcommand used."
             // -- https://ircv3.net/specs/extensions/chathistory#returned-message-notes
             Ok(backward_entries
                 .into_iter()
                 .rev()
-                .chain(forward_entries.into_iter()))
+                .chain(forward_entries.into_iter())
+                .flat_map(
+                    move |HistoryLogEntry {
+                              id: _,
+                              timestamp: _,
+                              source_event: _,
+                              details,
+                          }| match details {
+                        NetworkStateChange::NewMessage(update::NewMessage {
+                            message,
+                            source,
+                            target: _, // assume it's the same as the argument we got
+                        }) => Some(HistoricalEvent::Message {
+                            id: *message,
+                            message_type: crate::network::state::MessageType::Notice, // TODO
+                            source: "".to_string(),                                   // TODO
+                            source_account: None,                                     // TODO
+                            target,
+                            text: "".to_string(), // TODO
+                        }),
```

Comment:
@spb More blanks to fill

Reply:
At this point you need to look up the actual message by ID in the network state and use the contents to fill in these fields.

Reply:
I know but I was hoping you would do it for me :)

I'll get to it after I'm done with the rest if you're busy

Reply:
See 9731d9b0baab47bf9920adb960803a68e875e2dc for most of it